### PR TITLE
Features/user

### PR DESCRIPTION
**Pull request by Anastasiia-Svechnikova**
Feat: user profile logic an view components, angular material theme, all general interfaces, custom icon material service

### CODE REVIEW

File: src/index.html [hunk 1]
```diff
@@ -13,10 +13,15 @@
       href="favicon.ico" />
     <link
       rel="preconnect"
-      href="https://fonts.gstatic.com" />
+      href="https://fonts.googleapis.com" />
     <link
-      href="https://fonts.googleapis.com/css2?family=Roboto:wght@300;400;500&display=swap"
+      rel="preconnect"
+      href="https://fonts.gstatic.com"
+      crossorigin />
```
Comment (Las04ka) [inline src/index.html:20]: sort attributes

**Review by Las04ka — APPROVED** [review]

File: src/app/app.component.ts [hunk 2]
```diff
@@ -1,9 +1,14 @@
...
 @Component({
   selector: 'app-root',
   templateUrl: './app.component.html',
   styleUrls: ['./app.component.scss'],
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
-export class AppComponent {}
+export class AppComponent {
+  constructor(private iconsService: CustomIconService) {
+    this.iconsService.loadIcons();
```
Comment (AleaJactaEst) [inline src/app/app.component.ts:12]: looks quite complex for me, is there a better way for using material icons? haven't seen this before

File: src/app/auth/state/effects.ts [hunk 3]
```diff
@@ -34,10 +34,16 @@ export class AuthEffects {
               'Authorization',
               loginResponseData.tokens.accessToken,
             );
-            this.router.navigateByUrl('');
+            this.localStorageService.saveData(
```
Comment (AleaJactaEst) [inline src/app/auth/state/effects.ts:37]: why do we need userId in LS? 

probably we should have route on API like GET /me or /user

File: src/app/main/main.module.ts [hunk 4]
```diff
@@ -1,22 +1,53 @@
 import { NgModule } from '@angular/core';
-import { CommonModule } from '@angular/common';
 import { RouterModule, Routes } from '@angular/router';
 import { StoreModule } from '@ngrx/store';
+import { EffectsModule } from '@ngrx/effects';
 
 import { MainReducer } from './state/reducer';
-import { MainPageComponent } from './components/main-page/main-page.component';
+import { NavigationModule } from '../navigation/navigation.module';
```
Comment (AleaJactaEst) [inline src/app/main/main.module.ts:7]: we need to decided to follow the rule of full path imports or not, up to you

File: src/app/main/main.module.ts [hunk 5]
```diff
@@ -1,22 +1,53 @@
...
-    component: MainPageComponent,
+    canActivate: [],
+    component: SiteLayoutComponent,
+    children: [
+      {
+        path: 'home',
+        component: HomeComponent,
+        data: { header: HeaderTitles.home },
+      },
+      {
```
Comment (AleaJactaEst) [inline src/app/main/main.module.ts:25]: i guess, everything that happens after login should be /admin-panel even user-profile, with same guards etc.. not sure, wdyt here?

File: src/app/shared/interfaces/user-details.ts [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+import { IUserWorkType } from './user-work-type';
+
+export interface IUserDetails {
+  id: number;
+  email: string;
+  password: string;
+  name: null | string;
+  surname: null | string;
+  positionDescription: null | string;
+  language: null | string;
```
Comment (AleaJactaEst) [inline src/app/shared/interfaces/user-details.ts:18]: do you have a rule for strict null check or smth liek that? Do we need it? As far as i now by default null allowed for all types

Comment (AleaJactaEst) [reply src/app/shared/interfaces/user-details.ts:18]: Also a lot's of field we probably can do optional

File: src/app/user/components/section-upper-bar/section-upper-bar.component.html [hunk 7]
```diff
@@ -0,0 +1,21 @@
+<div class="wrapper">
+  <h3 class="subtitle-accent">
+    {{ title }}
+  </h3>
+  <button
+    *ngIf="mode !== 'noControls'"
+    mat-icon-button
+    class="icon-btn"
+    aria-label="Edit content button"
+    (click)="openDialog()">
```
Comment (AleaJactaEst) [inline src/app/user/components/section-upper-bar/section-upper-bar.component.html:10]: same method on different buttons hmm.. 
maybe you can do some ng-cont.. to not duplicate if

File: src/app/user/components/section-upper-bar/section-upper-bar.ts [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+
+@Component({
+  selector: 'app-section-upper-bar',
+  templateUrl: './section-upper-bar.component.html',
+  styleUrls: ['./section-upper-bar.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class SectionUpperBarComponent
+  extends UnSubscriberComponent
+  implements OnInit
```
Comment (AleaJactaEst) [inline src/app/user/components/section-upper-bar/section-upper-bar.ts:23]: can be one line?

File: src/app/user/components/section-upper-bar/section-upper-bar.ts [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+  @Input() section!: UserProfileInfoSections;
+  @Input() mode!: 'full' | 'noControls';
+  title = '';
+
+  constructor(public dialog: MatDialog) {
+    super();
+  }
+
+  ngOnInit(): void {
+    if (this.section) {
```
Comment (AleaJactaEst) [inline src/app/user/components/section-upper-bar/section-upper-bar.ts:34]: is it make sense to have title on section level?

File: src/app/navigation/constants/constants.ts [hunk 10]
```diff
@@ -1,26 +1,26 @@
...
     iconName: 'dashboard',
     title: 'projects',
     accessRoles: [UserRoles.Admin, UserRoles.Guest],
   },
   {
-    path: '/admin',
+    path: ['/dashboard', 'admin'],
     iconName: 'bar_chart',
     title: 'admin',
     accessRoles: [UserRoles.Admin],
```
Comment (Las04ka) [inline src/app/navigation/constants/constants.ts:26]: I think admin has access to all routes, so mb make it more automated?

File: src/app/user/components/user-profile/user-bank-and-socials-info/user-bank-and-socials-info.component.ts [hunk 11]
```diff
@@ -0,0 +1,102 @@
...
+    map(
+      (user) =>
+        ({
+          upwork: user?.upwork,
+          github: user?.github,
+          linkedin: user?.linkedin,
+          telegramTag: user?.telegramTag,
+        } as IUserSocialLinksData),
+    ),
+  );
```
Comment (Las04ka) [inline src/app/user/components/user-profile/user-bank-and-socials-info/user-bank-and-socials-info.component.ts:54]: mb make it in selector

File: src/app/user/components/user-profile/user-personal-info/user-personal-info.component.html [hunk 12]
```diff
@@ -0,0 +1,65 @@
...
+          <mat-icon>edit</mat-icon>
+        </button>
+        <app-clipboard [data]="userData"></app-clipboard>
+      </div>
+      <p
+        *ngIf="userData.status"
+        class="detail">
+        Status:
+        <span [ngClass]="userData.status">
+          {{ userData.status === 'archived' ? 'Deactivated ' : 'Active ' }}
```
Comment (Las04ka) [inline src/app/user/components/user-profile/user-personal-info/user-personal-info.component.html:25]: there is whitespace

File: src/app/user/components/user-profile/user-work-info/user-work-info.component.html [hunk 13]
```diff
@@ -0,0 +1,28 @@
...
+    <button
+      mat-icon-button
+      class="icon-btn"
+      aria-label="Edit content button">
+      <mat-icon>edit</mat-icon>
+    </button>
+    <app-clipboard [data]="{workHistory}"></app-clipboard>
+  </div>
+  <div
+    *ngFor="let work of workHistory; let i = index"
```
Comment (Las04ka) [inline src/app/user/components/user-profile/user-work-info/user-work-info.component.html:13]: no index used

**Pull request merged**